File: library/src/test/java/me/proxer/library/enums/TagSubTypeTest.java
package me.proxer.library.enums;

import me.proxer.library.ProxerTest;
import org.junit.Test;

import java.io.IOException;

import static org.assertj.core.api.Assertions.assertThat;

public class TagSubTypeTest extends ProxerTest {

    @Test
    public void testDefault() throws IOException {
        final TagSubType tagSubType = api.moshi().adapter(TagSubType.class).fromJson("\"story\"");

        assertThat(tagSubType).isSameAs(TagSubType.STORY);
    }

    @Test
    public void testFallback() throws IOException {
        final TagSubType tagSubType = api.moshi().adapter(TagSubType.class).fromJson("\"xyz\"");

        assertThat(tagSubType).isSameAs(TagSubType.OTHER);
    }
}

File: library/src/test/kotlin/me/proxer/library/enums/TagSubTypeTest.kt
package me.proxer.library.enums

import com.serjltt.moshi.adapters.FallbackEnum
import com.squareup.moshi.Moshi
import org.amshove.kluent.shouldBe
import org.junit.jupiter.api.Test

/**
 * @author Ruben Gees
 */
class TagSubTypeTest {

    private val adapter = Moshi.Builder()
        .add(FallbackEnum.ADAPTER_FACTORY)
        .build()
        .adapter(TagSubType::class.java)

    @Test
    fun testDefault() {
        adapter.fromJson("\"story\"") shouldBe TagSubType.STORY
    }

    @Test
    fun testFallback() {
        adapter.fromJson("\"xyz\"") shouldBe TagSubType.OTHER
    }
}
